### PR TITLE
More blendmodes

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
Ready for review!

Example: https://www.pixiplayground.com/#/edit/Y3HeuBbbllstbzsfp5g8t

There's no `SUBTRACT` blendMode for canvas2d.

Some blendModes require extra `clip` on canvas2d.

Code of CanvasSpriteRenderer is improved, slightly.

Aliases: 

1.  `DST_OUT` aka `ERASE` is the best eraser, it also does not require `clip` on canvas2d!
2. `SRC_OVER` is our default normal blendMode.

### CODE REVIEW

File: packages/core/src/state/StateSystem.js [hunk 1]
```diff
@@ -71,10 +71,10 @@ export default class StateSystem extends System
         /**
          * Blend mode
          * @member {number}
-         * @default 17
+         * @default 20
          * @readonly
          */
-        this.blendMode = 17;
+        this.blendMode = 20;
```
Comment (bigtimebuddy) [inline packages/core/src/state/StateSystem.js]: Change this to a reference to the BLEND_MODES.NONE

File: packages/core/src/state/utils/mapWebGLBlendModesToPixi.js [hunk 2]
```diff
@@ -43,5 +43,25 @@ export default function mapWebGLBlendModesToPixi(gl, array = [])
...
+    array[BLEND_MODES.DESTINATION_OVER] = [gl.ONE_MINUS_DST_ALPHA, gl.ONE, gl.ONE_MINUS_DST_ALPHA, gl.ONE];
+    array[BLEND_MODES.DESTINATION_IN] = [gl.ZERO, gl.SRC_ALPHA, gl.ZERO, gl.SRC_ALPHA];
+    array[BLEND_MODES.DESTINATION_OUT] = [gl.ZERO, gl.ONE_MINUS_SRC_ALPHA, gl.ZERO, gl.ONE_MINUS_SRC_ALPHA];
+    array[BLEND_MODES.DESTINATION_ATOP] = [gl.ONE_MINUS_DST_ALPHA, gl.SRC_ALPHA, gl.ONE_MINUS_DST_ALPHA, gl.SRC_ALPHA];
+
+    // SUBTRACT from flash
+    array[BLEND_MODES.SUBTRACT] = [gl.ONE, gl.ONE, gl.ONE, gl.ONE, gl.FUNC_REVERSE_SUBTRACT, gl.FUNC_ADD];
+
+    // SOURCE_IN:      21,
+    //     SOURCE_OUT:     22,
```
Comment (bigtimebuddy) [inline packages/core/src/state/utils/mapWebGLBlendModesToPixi.js]: Remove comments

**Comment by Dairnon** [comment]
With WEBGL2.0 we get more out of box support for other blend modes right? Stuff like screen? 

Or does that still require renderTexture/shader hacks.

**Comment by ivanpopelyshev** [comment]
Something is very wrong with canvas renderer blendModes, I'm working on it. WebGL ones are working.

**Comment by ivanpopelyshev** [comment]
According to example and masking, we have to add extra clip for canvas renderer to handle 5 of new blendmodes, both in Graphics and Sprite.

**Comment by ivanpopelyshev** [comment]
Unfortunately, use of `blendFuncSeparate` and extra `blendEq` affects performance: https://jsperf.com/blendmodefunc-test . I do not know what to do about it yet.

**Comment by ivanpopelyshev** [comment]
OK, optimization suggested by @GoodBoyDigital

File: packages/core/src/state/StateSystem.js [hunk 3]
```diff
@@ -240,14 +247,25 @@ export default class StateSystem extends System
...
         {
-            this.gl.blendFunc(mode[0], mode[1]);
+            gl.blendFunc(mode[0], mode[1]);
         }
         else
         {
-            this.gl.blendFuncSeparate(mode[0], mode[1], mode[2], mode[3]);
+            gl.blendFuncSeparate(mode[0], mode[1], mode[2], mode[3]);
+        }
+        if (mode.length === 6)
```
Comment (GoodBoyDigital) [inline packages/core/src/state/StateSystem.js]: can we stick this if inside the else above?

Comment (ivanpopelyshev) [reply packages/core/src/state/StateSystem.js]: no, because there's `else` branch for that if too

**Comment by ivanpopelyshev** [comment]
I looked through it one more time, seems clean.

**Pull request merged**